Kitware/dive
# Bugfix-877 Frame Alignment issues

Fixes #877 
fixes #909

I provided read access to this package for @subdavis  and @marySalvi : 

- https://viame.kitware.com/api/v1/item/61200b0a81830b81d0a57332/download
- This package includes a good60fps.mp4 file which contains corrected frames and a bad60fps.mp4 and bad60fps.mov files.
- All files have burnt in frame numbers and timestamps.  The included ReadMe.txt has the ffmpeg command to burn those in on other files if you want to.
- The index.html file will load up the good and bad mp4 files and allow you to advance through them frame by frame to show the behavior of DIVE and how it is offset between the two files.

The process for the fix requires reading in the ffprobe for duplicate `best_effort_timestamp_time` within the first 5 seconds of the video.  If there are duplicate frames it will then do a secondary process of encoding the video with `-ss 0` in an attempt to fix the audio alignment problem. 
NOTE:  Main issue is that before and after the fix the ffprobe `best_effort_timestamp_time` outputs look identical so I have no way to check from the command line if the fix worked.

Testing procedure:
I provided both a good and bad 60fps files so it would be easier to upload the bad60fps.mp4 and set the frame rate to 60fps.  In that instance one frame click forward should equal one frame on the burnt in video frame counter.

Notes about implementation:
Tried to move all of this extra logic into separate files to not pollute existing items.
On Desktop Check Media needs to do the ffprobe because we refrain from checking properly encoded mp4 videos.  This is why I specifically included a bad60fps.mp4 so you can test desktop importing and that it will kick off the job.
